Solutions/Menes.PetStore.Specs/Steps/Steps.cs

```diff
@@ -0,0 +1,209 @@
+﻿// <copyright file="Steps.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Menes.PetStore.Specs.Steps
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+    using NUnit.Framework;
+    using TechTalk.SpecFlow;
+
+    [Binding]
+    public class Steps
+    {
+        private static readonly HttpClient HttpClient = HttpClientFactory.Create();
+
+        private static readonly Uri BaseUri = new Uri("http://localhost:7071");
+
+        private readonly ScenarioContext scenarioContext;
+
+        public Steps(ScenarioContext scenarioContext)
+        {
+            this.scenarioContext = scenarioContext;
+        }
+
+        [When("I request a list of pets")]
+        public Task WhenIRequestAListOfPets()
+        {
+            return this.SendGetRequest("/pets");
+        }
+
+        [When("I request the pet with Id (.*)")]
+        public Task WhenIRequestThePetWithId(int petId)
+        {
+            return this.SendGetRequest($"/pets/{petId}");
+        }
+
+        [When("I request that a new pet be created")]
+        public Task WhenIRequestThatANewPetBeCreated(Table table)
+        {
+            if (table.RowCount != 1)
+            {
+                Assert.Fail("Exactly one row of pet details must be supplied.");
+            }
+
+            TableRow petRow = table.Rows[0];
+
+            // We want to be able to send invalid values for Id to test that the validation is working correctly.
+            // But if the value is a valid integer, we need to make sure it's passed correctly, so...
+            string? idFieldValue = this.ParseStringValue(petRow["Id"]);
+            object? id = long.TryParse(idFieldValue, out long idAsLong)
+                ? (object?)idAsLong
+                : idFieldValue;
+
+            var pet = new
+            {
+                id = id,
+                name = this.ParseStringValue(petRow["Name"]),
+                tag = this.ParseStringValue(petRow["Tag"]),
+                size = this.ParseStringValue(petRow["Size"]),
+            };
+
+            return this.SendPostRequest("/pets", pet);
+        }
+
+        [Given("I have requested a list of pets with a limit of (.*)")]
+        [When("I request a list of pets with a limit of (.*)")]
+        public Task WhenIRequestAListOfPetsWithALimitOf(int limit)
+        {
+            return this.SendGetRequest($"/pets?limit={limit}");
+        }
+
+        [When("I request a list of pets using the value of '(.*)' from the previous response object")]
+        public Task WhenIRequestAListOfPetsUsingTheValueOfFromThePreviousResponseObject(string propertyPath)
+        {
+            JToken token = this.GetExpectedTokenFromResponseObject(propertyPath);
```

Comment:
I'm wondering if this `GetExpectedTokenFromResponseObject` is appropriately named. I suspect that was a good name for what it was originally for, but now you're using it for several things I'm not sure the name correctly covers all of them. Here you seem to be using it to extract a token that actually came from a response (so "expected" isn't right here), which you then pass into something else.

Reply:
Better?

Reply:
Much!